ports/models.py

```diff
@@ -28,6 +33,236 @@ class Port(models.Model):
 
     objects = PortManager()
 
+    @classmethod
+    def load(cls, data):
+        def open_portindex_json(path):
+            with open(path, "r") as file:
+                ports = json.load(file)
+            return ports
+
+        # Add All the Categories to the Database using bulk_create
+        def load_categories_table(ports):
+            categories = set()
+            for port in ports:
+                try:
+                    for category in port['categories']:
+                        categories.add(Category(name=category))
+                except KeyError:
+                    pass
+            batch = list(categories)
+            Category.objects.bulk_create(batch)
+            return
+
+        def load_ports_and_maintainers_table(ports):
+            for port in ports:
+
+                # Add Ports to the Database One-by-One
+                new_port = Port()
+                try:
+                    new_port.name = port['name']
```

Comment:
So, does this mean `name` attribute is not necessary? Same for `portindex` and `version`.

I may have missed conversation around this but if there is something, it'd help if there's a comment on top since this is not trivial IMO. Weren't we having conversations around making the `name` as the key?

Won't this need to specify `null=True` or equivalent in the `Port` model above if that is the case indeed?

Reply:
Maybe I have implemented it in a wrong way, but the idea is to not add the port if any of these fields (name, version, portdir) are missing. That is why I added `continue` in the `except` block. Is something wrong with the code?

There should not be cases where any of these keys is missing. But in case if there is one- we would not want to add the port because something is wrong with it.

Reply:
Okay, my bad misinterpreting the `continue` with `pass`, thank you.